File: src/main/java/com/back/together02be/global/security/RefreshTokenCookieManager.java
package com.back.together02be.global.security;

import jakarta.servlet.http.Cookie;
import jakarta.servlet.http.HttpServletResponse;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

@Component
public class RefreshTokenCookieManager {

    public static final String COOKIE_NAME = "refreshToken";

    private static final String COOKIE_PATH = "/";
    private static final String COOKIE_DOMAIN = "localhost";
    private static final String SAME_SITE = "Strict";

    private final int refreshExpireSeconds;

    public RefreshTokenCookieManager(@Value("${jwt.refresh-expire-seconds}") int refreshExpireSeconds) {
        this.refreshExpireSeconds = refreshExpireSeconds;
    }

    public void addRefreshTokenCookie(HttpServletResponse response, String refreshToken) {
        Cookie cookie = createBaseCookie(refreshToken);
        cookie.setMaxAge(refreshExpireSeconds);
        response.addCookie(cookie);
    }

    public void deleteRefreshTokenCookie(HttpServletResponse response) {
        Cookie cookie = createBaseCookie("");
        cookie.setMaxAge(0);
        response.addCookie(cookie);
    }

    private Cookie createBaseCookie(String value) {
        Cookie cookie = new Cookie(COOKIE_NAME, value);
        cookie.setPath(COOKIE_PATH);
        cookie.setHttpOnly(true);
        cookie.setDomain(COOKIE_DOMAIN);
        cookie.setSecure(true);
        cookie.setAttribute("SameSite", SAME_SITE);
        return cookie;
    }
}

File: src/main/java/com/back/together02be/global/security/RefreshTokenCookieManager.kt
package com.back.together02be.global.security

import jakarta.servlet.http.Cookie
import jakarta.servlet.http.HttpServletResponse
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component

@Component
class RefreshTokenCookieManager(
    @Value("\${jwt.refresh-expire-seconds}")
    private val refreshExpireSeconds: Int
) {

    fun addRefreshTokenCookie(response: HttpServletResponse, refreshToken: String) {
        val cookie = createBaseCookie(refreshToken)
        cookie.maxAge = refreshExpireSeconds
        response.addCookie(cookie)
    }

    fun deleteRefreshTokenCookie(response: HttpServletResponse) {
        val cookie = createBaseCookie("")
        cookie.maxAge = 0
        response.addCookie(cookie)
    }

    private fun createBaseCookie(value: String): Cookie {
        return Cookie(COOKIE_NAME, value).apply {
            path = COOKIE_PATH
            isHttpOnly = true
            domain = COOKIE_DOMAIN
            secure = true
            setAttribute("SameSite", SAME_SITE)
        }
    }

    companion object {
        const val COOKIE_NAME = "refreshToken"

        private const val COOKIE_PATH = "/"
        private const val COOKIE_DOMAIN = "localhost"
        private const val SAME_SITE = "Strict"
    }
}
